flake.nix

```diff
@@ -159,6 +159,12 @@
               pkgs.ocamlPackages.mel
             ];
           };
+          slim-opam = with pkgs; mkShell {
```

Comment:
Could you leave a comment with a couple of sentences what this is for. We should probably do this for all the shells we define, but we can start here.

Reply:
Good idea. Added a description. Will send a separate PR for the others.